src/tools/msvc.jam

```diff
@@ -1624,7 +1624,7 @@ if [ MATCH (--debug-configuration) : [ modules.peek : ARGV ] ]
 .version-11.0-reg = "VisualStudio\\11.0\\Setup\\VC" ;
 .version-12.0-reg = "VisualStudio\\12.0\\Setup\\VC" ;
 .version-14.0-reg = "VisualStudio\\14.0\\Setup\\VC" ;
-.version-15.0-reg = "VisualStudio\\15.0\\Setup\\VC" ;
+.version-14.1-reg = "VisualStudio\\14.1\\Setup\\VC" ;
```

Comment:
They don't have register keys for VS2017. Should I just remove it?
How to support properly VS2017 path detection is  still being debated in #157